docs/proposals/workflow.md

```diff
@@ -0,0 +1,174 @@
+Example of generating installer from workflow configuration:
+```
+ftl_installer -w foo_workflow.yml -o foo_installer
+```
+
+Top level config keys:
+- config
+- variants
+- workflows
+- stages
+- steps
+- step_groups
+- roles
+
+
+Sample Workflow Definition
+```
+--- 
+config:
+  version: 2.2
+  migrations:
+  - migrate_v1_to_v2:
+    when: "{{ config_version | version_compare('lt', 2.0) }}"
```

Comment:
@abutcher added this as I was writing up what we discussed yesterday. I think it makes sense here to leverage the ansible concept of conditionals (and possibly loops). Thoughts?